packages/gatsby/src/schema/resolvers.js

```diff
@@ -0,0 +1,244 @@
+    .reduce((acc, fieldValue) => {
+      acc.push({
+        ...paginate(groupedResults[fieldValue], args),
+        field,
+        fieldValue,
+      })
+      return acc
+    }, [])
+}
+
+const paginate = (results, { skip = 0, limit }) => {
+  const count = results.length
+  const items = results.slice(skip, limit && skip + limit)
+
+  // const pageCount = limit
+  //   ? Math.ceil(skip / limit) + Math.ceil((count - skip) / limit)
+  //   : skip
+  //   ? 2
+  //   : 1
+  // const currentPage = limit ? Math.ceil(skip / limit) + 1 : skip ? 2 : 1 // Math.min(currentPage, pageCount)
+  // // const hasPreviousPage = currentPage > 1
+  const hasNextPage = skip + limit < count // currentPage < pageCount
+
+  return {
+    totalCount: items.length,
+    edges: items.map((item, i, arr) => {
+      return {
+        node: item,
+        next: arr[i + 1],
+        previous: arr[i - 1],
+      }
+    }),
+    pageInfo: {
+      hasNextPage,
+      // currentPage,
+      // hasPreviousPage,
+      // itemCount: count,
+      // pageCount,
+      // perPage: limit,
+    },
+  }
+}
+
+const getValueAtSelector = (obj, selector) => {
+  const selectors = Array.isArray(selector) ? selector : selector.split(`.`)
+  return selectors.reduce((acc, key) => {
+    if (acc && typeof acc === `object`) {
+      if (Array.isArray(acc)) {
+        return acc.map(a => a[key]).filter(a => a !== undefined)
+      }
+      return acc[key]
+    }
+    return undefined
+  }, obj)
+}
+
+// FIXME: Handle array of arrays
+// Maybe TODO: should we check fieldValue *and* info.returnType?
+const link = ({ by, from }) => async (source, args, context, info) => {
+  const fieldValue = source && source[from || info.fieldName]
+
+  if (fieldValue == null || _.isObject(fieldValue)) return fieldValue
+  if (
+    Array.isArray(fieldValue) &&
+    // TODO: Do we have to look with fieldValue.some(v => isObject(v))?
+    (fieldValue[0] == null || _.isObject(fieldValue[0]))
+  ) {
+    return fieldValue
+  }
+
+  if (by === `id`) {
+    const [resolve, key] = Array.isArray(fieldValue)
+      ? [withPageDependencies(findByIds), `ids`]
+      : [withPageDependencies(findById), `id`]
+    return resolve()({
+      source,
+      args: { [key]: fieldValue },
+      context,
+      info,
+    })
+  }
+
+  const equals = value => {
+    return { eq: value }
+  }
+  const oneOf = value => {
+    return { in: value }
+  }
+  const operator = Array.isArray(fieldValue) ? oneOf : equals
+  args.filter = by.split(`.`).reduceRight((acc, key, i, { length }) => {
+    return {
+      [key]: i === length - 1 ? operator(acc) : acc,
+    }
+  }, fieldValue)
+
+  const returnType = getNullableType(info.returnType)
+  const type = getNamedType(returnType)
+  const possibleTypes = isAbstractType(type)
+    ? info.schema.getPossibleTypes(type)
+    : [type]
+
+  if (returnType instanceof GraphQLList) {
+    const results = await Promise.all(
+      possibleTypes.map(type =>
+        findMany(type.name)({
```

Comment:
Now that `link` is in the same file and does not use the `findMany` wrapped with `withPageDependencies` we need to do this ourselves